pyfr/integrators/dual/pseudo/__init__.py

```diff
@@ -1,25 +1,72 @@
 # -*- coding: utf-8 -*-
 
+from pkg_resources import resource_listdir
 import re
 
 from pyfr.integrators.dual.pseudo.pseudocontrollers import BaseDualPseudoController
-from pyfr.integrators.dual.pseudo.pseudosteppers import BaseDualPseudoStepper
+from pyfr.integrators.dual.pseudo.pseudosteppers import BaseDualPseudoStepper, DualDenseRKPseudoStepper
 from pyfr.integrators.dual.pseudo.multip import DualMultiPIntegrator
 from pyfr.util import subclass_where
 
 
+def register_tabulated_pseudo_steppers():
+    if hasattr(register_tabulated_pseudo_steppers, '_schemes'):
+        return
+
+    # Create subclasses for all tabulated dense schemes
+    rpaths = resource_listdir('pyfr.integrators', 'schemes')
+
+    register_tabulated_pseudo_steppers._schemes = schemes = []
+    for path in rpaths:
+        m = re.match(r'([a-zA-Z0-9\-~+]+)-s(\d+)-p(\d+)'
+                     r'(?:-sp(\d+))(?:-([e]+))?\.txt', path)
+        name = m.group(1)
+
+        attrs = {'pseudo_stepper_name': name, 'path': path}
+        attrs['_stepper_nfevals'] = int(m.group(2)) + 1
+        attrs['_pseudo_stepper_nregs'] = int(m.group(2))
+        attrs['_pseudo_stepper_order'] = int(m.group(3))
+        attrs['_pseudo_stepper_has_lerrest'] = bool(m.group(5))
+
+        if m.group(4):
+            attrs['pseudo_stepper_porder'] = int(m.group(4))
+            name += '_ps{}'.format(m.group(4))
+
+        schemes.append(type(name, (DualDenseRKPseudoStepper,), attrs))
+
+    register_tabulated_pseudo_steppers._has_been_called = True
+
+
+def get_pseudo_steppercls(name, porder):
+    for p in range(porder, -1, -1):
+        try:
+            pc = subclass_where(BaseDualPseudoStepper,
```

Comment:
return

Reply:
^

Reply:
Ah, now I got it. Pushing one more time.